knitout-to-kcode.js

```diff
@@ -80,7 +80,11 @@ const OP_MISS_FRONT_TUCK_BACK = { name:'OP_MISS_FRONT_TUCK_BACK', isBack:true };
 const OP_MISS_FRONT_MISS_BACK = { name:'OP_MISS_FRONT_MISS_BACK' };
 
 const OP_XFER_TO_BACK  = { name:'OP_XFER_TO_BACK', isFront:true };
-const OP_XFER_TO_FRONT = { name:'OP_XFER_TO_FRONT', isBack:true };
+const OP_XFER_TO_FRONT = { name: 'OP_XFER_TO_FRONT', isBack:true };
+
+// const OP_SPLIT_TO_BACK = { name:'OP_SPLIT_TO_BACK', isFront:true }; //*
+// const OP_SPLIT_TO_FRONT = { name: 'OP_SPLIT_TO_FRONT', isBack: true }; //*
+const OP_SPLIT = { name: 'OP_SPLIT', isFront:true, isBack:true };
```

Comment:
I agree, only thing is, since `split` is handled along with `knit` `tuck` in `passesToKCode()`, the existing code requires that the `OP` includes both `isFront:true` and `isBack:true` for both the front & back needles to be addressed. So, to make it `SPLIT_TO_BACK` and `SPLIT_TO_FRONT` instead, we'd either need to patch in extra stuff in `passesToKCode`, or make them effectively the same thing (redundant code), like so:
```js
const SPLIT_TO_BACK = { name:'SPLIT_TO_BACK', isFront:true, isBack:true };
const SPLIT_TO_FRONT = { name:'SPLIT_TO_FRONT', isFront:true, isBack:true };
```
See the following code from `passesToKCode`(starting at line 1609), notably `if (f in pass.slots && pass.slots[f].isFront)` and `if (b in pass.slots && pass.slots[b].isBack)`:
```js
for (let n = 0; n <= 252; ++n) {
	const f = frontNeedleToSlot(n);
	if (f in pass.slots && pass.slots[f].isFront) {
		kpass.FRNT += '-';
	} else {
		kpass.FRNT += '_';
	}
	const b = backNeedleToSlot(n, pass.racking);
	if (b in pass.slots && pass.slots[b].isBack) {
		kpass.REAR += '-';
	} else {
		kpass.REAR += '_';
	}
}
```

I'm going to leave it as is for now, but if you come up with a solution that aligns more with your preferences, feel free to change it.

Reply:
Good point. I'm actually surprised that this required no code changes to the output at all. Fancy stuff.

Being that you plan to keep things the same, probably removing the commented-out `SPLIT_TO_*` constants would be good. (And perhaps adding a comment to explain the meaning, like `//the type of OP_SPLIT is determined by the type of the pass that contains it.`)